subsys/testsuite/ztest/include/ztress.h

```diff
@@ -0,0 +1,185 @@
+ *
+ * @param handler User handler of type @ref ztress_handler.
+ *
+ * @param user_data User data passed to the @p handler.
+ *
+ * @param exec_cnt Number of handler executions to complete the test. If 0 then
+ * this is not included in completion criteria.
+ *
+ * @param preempt_cnt Number of preemptions of that context to complete the test.
+ * If 0 then this is not included in completion criteria.
+ *
+ * @param init_timeout Initial backoff time base (given in @ref k_timeout_t). It
+ * is adjusted during the test to optimize CPU load. The actual timeout used for
+ * sleeping is randomized.
+ */
+#define ZTRESS_THREAD(handler, user_data, exec_cnt, preempt_cnt, init_timeout) \
+	(ZTRESS_ID_THREAD, handler, user_data, exec_cnt, preempt_cnt, init_timeout)
+
+/** @brief User handler
+ *
+ * @retval true continue test.
+ * @retval false end given context.
+ */
+typedef bool (*ztress_handler)(void *user_data, uint32_t cnt, bool last, int prio);
+
+/** @internal Context structure. */
+struct ztress_context_data {
+	/* Handler. */
+	ztress_handler handler;
+
+	/* User data */
+	void *user_data;
+
+	/* Minimum number of executions to complete the test. */
+	uint32_t exec_cnt;
+
+	/* Minimum number of preemptions to complete the test. */
+	uint32_t preempt_cnt;
+
+	/* Initial timeout. */
+	k_timeout_t t;
+};
+
+/** @internal Macro for initializing context data. */
+#define Z_ZTRESS_GET_HANDLER_DATA(data) \
+	{ \
+		.handler = GET_ARG_N(2, __DEBRACKET data), \
+		.user_data = GET_ARG_N(3, __DEBRACKET data), \
+		.exec_cnt = GET_ARG_N(4, __DEBRACKET data), \
+		.preempt_cnt = GET_ARG_N(5, __DEBRACKET data), \
+		.t = GET_ARG_N(6, __DEBRACKET data), \
+	}
+
+/** @internal Macro for checking if provided context is a timer context. */
+#define Z_ZTRESS_HAS_TIMER(data, ...) \
+	GET_ARG_N(1, __DEBRACKET data)
+
+/** @brief Setup and run stress test.
+ *
+ * Macro setup all contexts and runs them till completion. Test runs until all
+ * completion requirements are met or until test timeouts (use @ref ztress_set_timeout
+ * to configure timeout) or until test is aborted (@ref ztress_abort).
+ *
+ * On test completion report is printed (@ref ztress_report is called internally).
+ *
+ * @param ... List of contexts. Contexts are configured using @ref ZTRESS_TIMER
+ * and @ref ZTRESS_THREAD macros. @ref ZTRESS_TIMER can only be used as the first
+ * argument. Each thread context has assigned priority. Priority is assigned in
+ * descending order (first context has the highest priority). Number of supported
+ * thread contexts is configurable in Kconfig.
+ */
+#define ZTRESS_EXECUTE(...) do { \
+	int has_timer = Z_ZTRESS_HAS_TIMER(__VA_ARGS__); \
+	struct ztress_context_data data[] = { \
+		FOR_EACH(Z_ZTRESS_GET_HANDLER_DATA, (,), __VA_ARGS__) \
+	}; \
+	size_t cnt = ARRAY_SIZE(data) - has_timer; \
+	int err = z_ztress_execute(has_timer ? &data[0] : NULL, &data[has_timer], cnt); \
+	zassert_equal(err, 0, "ztress_execute failed (err: %d)", err); \
+} while (0)
```

Comment:
Can you run clang-format on this patch? This doesn't seem to follow the style guide

Reply:
are you referring to lack of empty line after variable declaration? I think check-patch is not checking that in macros. Added.

Reply:
The position of the `\` character. Should be aligned at the right using spaces.

Reply:
done